tests/ui/associated-type-bounds/dont-imply-atb-in-closure-inference.rs

```diff
@@ -0,0 +1,21 @@
+//@ check-pass
```

Comment:
This is one of two user-facing consequences of this change, but I don't expect this to be replicatable with RPIT on stable (at least not easily, e.g. w/o a recursive call) since we eagerly replace opaques with infer vars in the old solver.